### PR TITLE
Use API key auth in templates by default

### PR DESCRIPTION
**Pull request by glahaye**
### Description
- Switch to API Key for auth method
- Merge the 2 main bicep files / factor our common code
- Allow passing of models to use in new deployments too

### Contribution Checklist
- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/copilot-chat-app/webapi/DeploymentTemplates/README.md [hunk 1]
```diff
@@ -125,6 +125,20 @@ check that you have correctly entered the values for the following settings:
...
+All of the server's endpoints other than the /probe one require authorization to access.
+By default, the deployment templates set up the server so that an API key is required to access its endpoints.
+
+AAD authentication and authorization can also be set up manually after the automated deployment is done.
+
+To view the API key required by your instance, access the page for your Semantic Kernel app service in the Azure portal.
+From that page, click on the "Configuration" item in the "Settings" section of the left pane. Then click on the text that reads "Hidden value.
+Click to show value" next to the "Authorization:ApiKey" setting.
+
+To authorize requests with the API key, it must be added as the value of an "x-api-key" header added to the requests.
```
Comment (Vivihung) [inline samples/apps/copilot-chat-app/webapi/DeploymentTemplates/README.md:139]: I don't see any SK app service logic changes that check the "x-api-key" header. Do we plan to have another PR to cover it?

**Review by Vivihung — COMMENTED** [review]
Thanks for the API key auth update in deployment scripts. Was wondering where is the app services change to validate "x-api-key" header of every request.

File: samples/apps/copilot-chat-app/webapi/DeploymentTemplates/README.md (hunk 1 above)
Comment (glahaye) [reply samples/apps/copilot-chat-app/webapi/DeploymentTemplates/README.md:139]: Correct.

**Review by hathind-ms — APPROVED** [review]
